state_machine/applications/flight/Tasks/radio.py

```diff
@@ -0,0 +1,103 @@
+            if heard_something:
+                response = self.cubesat.radio.receive(keep_listening=True, with_ack=ANTENNA_ATTACHED)
+                if response is not None:
+                    self.debug(f'Recieved msg "{response}", RSSI: {self.cubesat.radio.last_rssi - 137}')
+                    # Processing recieved messages goes here
+                    #  - Execute commands
+                    #  - Mark messages as received (and remove from tq)
+
+                    # header = response[0]
+                    response = response[1:]  # remove the header byte
+
+                    # Begin Old Beacon Task Code
+                    if len(response) >= 6:
+                        if not ANTENNA_ATTACHED:
+                            self.debug('Antenna not attached. Skipping over-the-air command handling')
+                        else:
+                            if response[:4] == self.super_secret_code:
+                                cmd = bytes(response[4:6])  # [pass-code(4 bytes)] [cmd 2 bytes] [args]
+                                cmd_args = None
+                                if len(response) > 6:
+                                    self.debug('command with args', 2)
+                                    try:
+                                        cmd_args = response[6:]  # arguments are everything after
+                                        self.debug(f'cmd args: {cmd_args}', 2)
+                                    except Exception as e:
+                                        self.debug(f'arg decoding error: {e}', 2)
+                                if cmd in cdh.commands:
+                                    try:
+                                        if cmd_args is None:
+                                            self.debug(f'running {cdh.commands[cmd]} (no args)')
+                                            self.cmd_dispatch[cdh.commands[cmd]](self)
+                                        else:
+                                            self.debug(f'running {cdh.commands[cmd]} (with args: {cmd_args})')
+                                            self.cmd_dispatch[cdh.commands[cmd]](self, cmd_args)
+                                    except Exception as e:
+                                        self.debug(f'something went wrong: {e}')
+                                        self.cubesat.radio.send(str(e).encode())
+                                else:
+                                    self.debug('invalid command!')
+                                    self.cubesat.radio.send(b'invalid cmd' + response[4:])
+                    # End Old Beacon Task Code
+            else:
+                self.debug('No packets received')
+        elif should_transmit():
+            msg = tq.peek()
+            packet, with_ack = msg.packet()
+            self.debug(f'Transmission Queue {tq.queue}')
+            short_packet = str(packet)[:20] + "...." if len(packet) > 23 else packet
```

Comment:
also, I could be misunderstanding this, but if msg is of type NaiveMessage, isn't msg.packet() already created within a given length? Why are we further shortening this packet later with short_packet? Additionally, assuming that packet is longer than 23 characters, how are we keeping track of the bytes after our "cursor" (packet[20:] in this case)?

Reply:
short_packet is only used when printed to the console.

Reply:
renamed to debug_packet to make this more clear